File: 112-path-sum/path-sum.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution {
    public boolean hasPathSum(TreeNode root, int targetSum) {
        if(root == null) return false;
        if(root.left == null && root.right == null)  {
            if(targetSum - root.val == 0) return true;
            else return false;
        }
        boolean isLeft = hasPathSum(root.left, targetSum - root.val);
        if(isLeft == true) return true;
        boolean isRight = hasPathSum(root.right, targetSum - root.val);
        return isRight;
    }
}

File: 112-path-sum/path-sum.js
/**
 * Definition for a binary tree node.
 * function TreeNode(val, left, right) {
 *     this.val = (val===undefined ? 0 : val)
 *     this.left = (left===undefined ? null : left)
 *     this.right = (right===undefined ? null : right)
 * }
 */
/**
 * @param {TreeNode} root
 * @param {number} targetSum
 * @return {boolean}
 */
var hasPathSum = function(root, targetSum) {
    if(!root) return false;
    let isFound = false;
    const travarse = (curr, sum) => {
        if(!curr.left && !curr.right) {
            if(isFound === false) isFound = sum + curr.val == targetSum;
            console.log(curr, sum, isFound, sum+curr.val);
        }
        curr.left && travarse(curr.left, sum + curr.val)
        curr.right && travarse(curr.right, sum + curr.val);
    }

    travarse(root, 0);
    return isFound;
};